manageiq-operator/controllers/manageiq_controller.go

```diff
@@ -147,10 +151,83 @@ func (r *ManageIQReconciler) Reconcile(ctx context.Context, request ctrl.Request
 	if e := r.manageApplicationResources(miqInstance); e != nil {
 		return reconcile.Result{}, e
 	}
+	if err := r.updateManageIQStatus(miqInstance); err != nil {
+		reqLogger.Error(err, "Failed setting ManageIQ status")
+		return reconcile.Result{}, err
+	}
 
 	return reconcile.Result{}, nil
 }
 
+func (r *ManageIQReconciler) updateManageIQStatus(cr *miqv1alpha1.ManageIQ) error {
+	miqInstance := &miqv1alpha1.ManageIQ{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: cr.Name, Namespace: cr.Namespace}, miqInstance)
+	if err != nil {
+		logger.Error(err, "Error getting cluster cr")
+		return err
+	}
+
+	// update status versions
+	r.reportOperatorVersions(miqInstance)
+
+	// update status condition
+	deployments := []string{"httpd", "memcached", "orchestrator", "postgresql"}
+	for _, deploymentName := range deployments {
+		if object := FindDeployment(cr, r.Client, deploymentName); object != nil {
+			deploymentStatusConditions := object.Status.Conditions
+
+			if typeReplicaFailure := FindDeploymentStatusCondition(deploymentStatusConditions, appsv1.DeploymentReplicaFailure); typeReplicaFailure != nil {
+				conditionMessage := fmt.Sprintf("[%s] %s", typeReplicaFailure.Type, typeReplicaFailure.Message)
+				r.reportStatusCondition(miqInstance, conditionMessage, typeReplicaFailure.Reason, metav1.ConditionStatus(typeReplicaFailure.Status), deploymentName)
+			} else if typeProgressingCondition := FindDeploymentStatusCondition(deploymentStatusConditions, appsv1.DeploymentProgressing); typeProgressingCondition != nil {
+				conditionMessage := fmt.Sprintf("[%s] %s", typeProgressingCondition.Type, typeProgressingCondition.Message)
+				r.reportStatusCondition(miqInstance, conditionMessage, typeProgressingCondition.Reason, metav1.ConditionStatus(typeProgressingCondition.Status), deploymentName)
+			} else if typeAvailableCondition := FindDeploymentStatusCondition(deploymentStatusConditions, appsv1.DeploymentAvailable); typeAvailableCondition != nil {
+				conditionMessage := fmt.Sprintf("[%s] %s", typeAvailableCondition.Type, typeAvailableCondition.Message)
+				r.reportStatusCondition(miqInstance, conditionMessage, typeAvailableCondition.Reason, metav1.ConditionStatus(typeAvailableCondition.Status), deploymentName)
+			}
```

Comment:
Is it possible to do something like this?
```suggestion
for i, conditionType := range [appsv1.DeploymentReplicaFailure, appsv1.DeploymentProgressing, appsv1.DeploymentAvailable] {
  for i, condition := range conditions {
    if condition.Type == conditionType {
      r.reportStatusCondition(miqInstance, fmt.Sprintf("[%s] %s", condition.Type, condition.Message), condition.Reason, metav1.ConditionStatus(condition.Status), deploymentName)
    }
  }
}
```

Reply:
Had a call with Brandon. Also discussed for loop approach.
After discussion we finalize on if elseif approach